### ISSUE TITLE
Login with username

### ISSUE THREAD

**Issue by elharo** [tootsuite/mastodon#6976]
RFE: allow logging in with username as well as email address. 

* * * *

- [ X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by nightpool** [comment]
why?

**Comment by nynhex** [comment]
@nightpool I agree with your `why?` however, a lot of systems are using the dual-auth strategy (email or password).  This is relatively simple t do with devise, and I've already started on an implementation in my fork to introduce this if anyone is interested.  

It's more of a cosmetic change than anything and probably not worth wasting cycles on.  I'm just doing it for fun and since I've done it before on other platforms in Rails.

IMHO, I don't think this is worth surfacing even with a low priority.  There are too many issues raised on core to even touch this.  Just my .02c.

If collaborators and maintainers decide that this feature is worthwhile, I'll be happy to share my work and PR.

cc: @elharo

**Comment by elharo** [comment]
Because doing otherwise confuses users. That is, it confused and surprised me. I'd be willing to bet I'm not the only one either.

Many sites do only use email for login, but these sites almost universally don't ask the user to pick a username too. My expectation as a user is that when I choose a username (almost always the same six character string I've been using since the early 90s) that's what I put in the login box. 

In this case, I had tried that several times first assuming I got the password wrong, and then that the site must be broken, or there was a time delay on new accounts or some such thing, before I finally noticed that I should try the email. 

You shouldn't get rid of the email login, of course. Both can happily coexist.

**Comment by elharo** [comment]
Also in this specific case a lot of folks are coming from and experience with Twitter which does allow login by username.

**Comment by mal0ki** [comment]
Since your email isn't displayed anywhere (except on the admin side), having you login with your email rather than account name does limit the possible attack vectors for compromising accounts.

**Comment by jakimfett** [comment]
oi @nynhex, what happened to the forked code you mentioned?

> reason for my interest:  
> I'm currently locked out of one of my fediverse accounts because I cannot remember which of my approximately thirty possible email addresses I created the acount under for which project.

Mind linking me to the area you were tinkering in if the code is no longer floating around?

**Comment by rugk** [comment]
shh bot
